pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -311,7 +311,7 @@ private SuccessResponse uploadSegment(FormDataMultiPart multiPart, boolean enabl
       // Validate segment
       SegmentValidatorResponse segmentValidatorResponse =
           new SegmentValidator(_pinotHelixResourceManager, _controllerConf, _executor, _connectionManager,
-              _controllerMetrics, _controllerLeadershipManager)
+              _controllerMetrics, _leadControllerManager)
```

Comment:
Can we just get the current isLeader boolean for this table and pass it down instead of passing _leadControllerManager all the way down to execute one restli call? It will be cleaner to have smaller number of classes dependent on lead controller manager.

Reply:
Done.

Reply:
thanks